Content.Server/NPC/Systems/NPCUseActionOnTargetSystem.cs

```diff
@@ -60,7 +77,26 @@ public override void Update(float frameTime)
             if (!htn.Blackboard.TryGetValue<EntityUid>(comp.TargetKey, out var target, EntityManager))
                 continue;
 
-            TryUseTentacleAttack((uid, comp), target);
+            TryUseAction((uid, comp), target);
+        }
+    }
+
+    private EntityUid ChooseAction(Dictionary<EntityUid, float> actions)
+    {
+        List<EntityUid> keys = actions.Keys.ToList();
+        List<float> values = actions.Values.ToList();
+        EntityUid defaultToReturn = keys[values.IndexOf(values.Max())];
+
+        for (int i = 0; i < actions.Count; i++)
+        {
```

Comment:
WeightRandom system, which is already implemented in EntityTable (I'm not sure), will fit here
see if you can't reuse it

Reply:
> EntityTable

Not in EntityTable